ocpp-transport-soap/src/main/kotlin/com/izivia/ocpp/http/HttpServer.kt

```diff
@@ -0,0 +1,108 @@
+package com.izivia.ocpp.http
+
+import com.izivia.ocpp.operation.information.ChargingStationConfig
+import com.izivia.ocpp.operation.information.RequestMetadata
+import com.izivia.ocpp.soap.OcppSoapParser
+import com.izivia.ocpp.soap.ResponseSoapMessage
+import com.izivia.ocpp.transport.OcppVersion
+import com.izivia.ocpp.transport.ServerTransport
+import org.http4k.contract.ContractRoute
+import org.http4k.contract.contract
+import org.http4k.contract.div
+import org.http4k.core.HttpHandler
+import org.http4k.core.Method.POST
+import org.http4k.core.Request
+import org.http4k.core.Response
+import org.http4k.core.Status.Companion.NOT_FOUND
+import org.http4k.core.Status.Companion.OK
+import org.http4k.lens.Path
+import org.http4k.server.Http4kServer
+import org.http4k.server.Undertow
+import org.http4k.server.asServer
+import org.slf4j.LoggerFactory
+import java.util.*
+import kotlin.reflect.KClass
+
+class HttpServer(
+    private val ocppVersion: OcppVersion,
+    private val port: Int,
+    private val path: String,
+    private val ocppSoapParser: OcppSoapParser,
+    private val newMessageId: () -> String = { UUID.randomUUID().toString() },
+    private var server: Http4kServer? = null
```

Comment:
if we provide a server, the port is not used. So it would be cleaner to have 2 distinct constructors or factory methods.